### PR TITLE
msvc: set warning level to /W4 globally

### PR DESCRIPTION
**Pull request by danilaml**
And fix a bunch of warnings

### CODE REVIEW

File: src/core/hle/service/ssl/ssl_backend_schannel.cpp [hunk 1]
```diff
@@ -477,7 +477,8 @@ class SSLConnectionBackendSchannel final : public SSLConnectionBackend {
             return ResultInternalError;
         }
         PCCERT_CONTEXT some_cert = nullptr;
-        while ((some_cert = CertEnumCertificatesInStore(returned_cert->hCertStore, some_cert))) {
+        while ((some_cert = CertEnumCertificatesInStore(returned_cert->hCertStore, some_cert)) !=
+               0) {
```
Comment (liamwhite) [inline src/core/hle/service/ssl/ssl_backend_schannel.cpp:481]:
```suggestion
               nullptr) {
```

File: src/core/hle/kernel/k_capabilities.cpp [hunk 2]
```diff
@@ -156,7 +156,7 @@ Result KCapabilities::MapIoPage_(const u32 cap, KPageTable* page_table) {
     const u64 phys_addr = MapIoPage{cap}.address.Value() * PageSize;
     const size_t num_pages = 1;
     const size_t size = num_pages * PageSize;
-    R_UNLESS(num_pages != 0, ResultInvalidSize);
+    // R_UNLESS(num_pages != 0, ResultInvalidSize);
```
Comment (liamwhite) [inline src/core/hle/kernel/k_capabilities.cpp:159]:
```suggestion
```

**Review by liamwhite — APPROVED** [review]
looks fine, but I don't daily drive windows

**Comment by danilaml** [comment]
@liamwhite do you think some else needs to lok at this? Perhaps @lioncash since they touched the warnigns last?

It's mainly just to scratch my itch of having so mcuh `D9025 : overriding '/W3' with '/W4'` and `D9025 : overriding '/Zi' with '/Z7'` warnings, that it was impossible to see anything usefull.

**Pull request merged**